subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BuildVerdictPluginTest.kt

```diff
@@ -0,0 +1,194 @@
+package com.avito.android.build_verdict
+
+import com.avito.android.build_verdict.BuildFailureListener.Companion.buildVerdictFileName
+import com.avito.test.gradle.TestProjectGenerator
+import com.avito.test.gradle.dir
+import com.avito.test.gradle.gradlew
+import com.avito.test.gradle.kotlinClass
+import com.avito.test.gradle.module.AndroidAppModule
+import com.google.common.truth.Truth
+import com.google.gson.GsonBuilder
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+class BuildVerdictPluginTest {
+
+    @field:TempDir
+    lateinit var temp: File
+    private val gson = GsonBuilder().create()
+
+    @Test
+    fun `When kotlin compilation fails Then build-verdict contains kotlin compile info`() {
+        generateProject {}
+
+        File(temp, "app/src/main/kotlin").kotlinClass("Uncompiled") { "incorrect syntax" }
+        val result = gradlew(
+            temp,
+            "assembleDebug",
+            expectFailure = true
+        )
+        result.assertThat()
+            .buildFailed()
+
+        assertBuildVerdict(
+            failedTask = "compileDebugKotlin",
+            errorOutput = "e: /private$temp/app/src/main/kotlin/Uncompiled.kt: (1, 1): Expecting a top level declaration\n" +
+                "e: /private$temp/app/src/main/kotlin/Uncompiled.kt: (1, 11): Expecting a top level declaration"
+        )
+    }
+
+    @Test
+    fun `When kapt stubs generating fails Then build-verdict contains kapt info`() {
```

Comment:
`when` `what` `if` blocks should be separated with ` - `

Reply:
We have already discussed that topic and I didn't get why this pattern is bad
I could change the naming for you but if you want me to make this in the future I propose to create documentation about test naming with motivation and discussed all pros and cons there

Reply:
it's just about consistency; we need single pattern for all tests in repository; if you're introducing new one - make sure it's changed across repo
i dont' mind one or another way, i'm just for single pattern, it's easier to read